src/main/java/cn/nukkit/Player.java

```diff
@@ -1434,7 +1434,7 @@ protected void handleMovement(Vector3 clientPos) {
         double distance = clientPos.distanceSquared(this);
         if (distance > 128) {
             invalidMotion = true;
-        } else if (!this.level.isChunkGenerated(clientPos.getChunkX(), clientPos.getChunkX())) {
+        } else if (this.chunk != null && !this.level.isChunkGenerated(clientPos.getChunkX(), clientPos.getChunkZ())) {
```

Comment:
You shouldn't need the null check as `Level#isChunkGenerated(int, int)` already checks this.